### ISSUE TITLE
Repo with CodeQL analysis still receives pull requests

### ISSUE THREAD

**Issue by jorsmatthys** [NickLiffen/ghas-enablement#65]
Hello,

After some testing my colleague noticed an issue with the pull requests for the CodeQL. It looks like a bug. 

**Expected behaviour**

> This script only returns repositories where CodeQL results have not already been uploaded to code scanning. If any CodeQL results have been uploaded to a repositories code scanning feature, that repository will not be returned to this list. The motivation behind this is not to raise pull requests on repositories where CodeQL has already been enabled.

**Actual behaviour**

Initially everything works as expected, ghas-enablement creates a pull request on our repository to add a CodeQL analysis workflow, we merge it, the CodeQL scan runs, but then on the next run of ghas-enablement, a new pull-request seems to get created. 

**Test environment**
* Linux github runner
* Version 3.3.0 of the code

**Further details**

As far as I can tell from searching on the code, the code that was foreseen to implement the expected behaviour is not yet being triggered (the code in `checkCodeQLEnablement.ts`). 

There are also some duplicate types that can probably be removed, as they do not seem to be used.

```
export type listCodeScanningParameters =
  Endpoints["GET /repos/{owner}/{repo}/code-scanning/analyses"]["parameters"];

export type listCodeScanningResponse =
  Endpoints["GET /repos/{owner}/{repo}/code-scanning/analyses"]["response"];
```

I suppose we need to adjust the code so that this code gets called and perhaps also supplement this check (someone might remove the codeql-analysis after initial results were published). 

Not sure how to label an issue as being a bug or feature :) perhaps only you can do that @NickLiffen

**Comment by NickLiffen** [comment]
@jorsmatthys yup I was able to replicate this 😢 This logic got removed in one of the releases as it was burning through API calls, this shouldn't be too hard to re-introduce, I just need to find a way to do it in an optimal way 👍 

I am on the case for this and will keep you posted 👍 this is a bug/feature 😂 I will get on this asao 💯 hopefully Thursday this week 👍

**Comment by maartengryp-liantis** [comment]
@NickLiffen any update on this issue?

**Comment by eglenimbus** [comment]
@NickLiffen do you have any news here ?

**Comment by NickLiffen** [comment]
This should now be fixed by #75, i will draft a new release now ❤️ Thanks for your patience 🙇
